src/ui/camera.js

```diff
@@ -957,8 +957,8 @@ class Camera extends Evented {
         this._ease((k) => {
             // s: The distance traveled along the flight path, measured in ρ-screenfuls.
             const s = k * S;
-            const scale = 1 / w(s);
-            tr.zoom = k === 1 ? zoom : startZoom + tr.scaleZoom(scale);
+            const scale = k === 1 ? tr.zoomScale(zoom - startZoom) : 1 / w(s);
+            tr.zoom = startZoom + tr.scaleZoom(scale);
```

Comment:
Let's keep the old `k === 1` condition here because zoomScale+scaleZoom may introduce loss of floating point precision.

Reply:
Good point. I will update the PR with your recommendation.